app/poros/direction.rb

```diff
@@ -0,0 +1,14 @@
+class Direction
+  attr_reader :name,
+              :travel_time,
+              :directions
+
+  def initialize(data)
+    @distance = data[:distance]
+    @travel_time = data[:formattedTime]
+    @directions = data[:legs].first[:maneuvers].map do |direction|
+      direction[:narrative]
+    end
```

Comment:
This should be it's own method called `directions` - if you're getting an undefined method `.first` for nilclass, that means that at least some of the time `data[:legs]` is returning a `nil`.